docs/source/extensions/entry_points.rst

```diff
@@ -0,0 +1,44 @@
+************
+Entry Points
+************
+
+Lightning supports registering Trainer callbacks directly through
+`Entry Points <https://setuptools.pypa.io/en/latest/userguide/entry_point.html>`_. Entry points allow an arbitrary
+package to include callbacks that the Lightning Trainer can automatically use, without you having to add them
+to the Trainer manually. This is useful in production environments where it is common to provide specialized monitoring
+and logging callbacks globally for every application.
+
+Here is a callback factory function that returns two special callbacks:
+
+.. code-block:: python
+    :caption: factories.py
+
+    def my_custom_callbacks_factory():
+        return [MyCallback1(), MyCallback2()]
+
+If we make this `factories.py` file into an installable package, we can define an **entry point** for this factory function.
+Here is a minimal example of the `setup.py` file for the package `my-package`:
+
+.. code-block:: python
+    :caption: setup.py
+
+    from setuptools import setup
+
+    setup(
+        name="my-package",
+        version="0.0.1",
+        entry_points={
```

Comment:
shall we then also add that PL need to be install requirements
```suggestion
        install_requires=["pytorch-lightning"],
        entry_points={
```

Reply:
It's not strictly required but I think it's good to mention